core/src/main/java/org/apache/calcite/rel/core/Sort.java

```diff
@@ -124,11 +126,20 @@ public abstract Sort copy(RelTraitSet traitSet, RelNode newInput,
 
   @Override public @Nullable RelOptCost computeSelfCost(RelOptPlanner planner,
       RelMetadataQuery mq) {
+    double rowCount = mq.getRowCount(this);
+    if (collation.getFieldCollations().isEmpty()) {
+      return planner.getCostFactory().makeCost(rowCount, 0, 0);
+    }
+    int offsetValue = offset == null || offset instanceof RexDynamicParam ? 0
+        : RexLiteral.intValue(offset);
```

Comment:
`RexLiteral.intValue` would do narrowing conversion, so it would result negative values for values that exceed `Integer.MAX_VALUE`

Reply:
I agree. I have try to add a RexLiteral.longValue method, but found there are too many place use RexLiteral.intValue, such as RelMdRowCount.

Reply:
I have logged https://issues.apache.org/jira/browse/CALCITE-4531 for `intValue` deprecation.
I am afraid `longValue` won't solve the problem since the value might exceed `Long.MAX_VALUE` as well.